contracts/safe-hts-precompile/SafeOperations.sol

```diff
@@ -6,6 +6,9 @@ import "./SafeHTS.sol";
 
 contract SafeOperations {
 
+    event TokenCreated(address tokenAddress);
```

Comment:
Might it be better to move safe contract under `hts-precompile/examples/safe`

Reply:
I think the safe contract should be located close to the SafeHTS library because it's meant to showcase its usage.
Keeping that in mind an alternative structure to the current one could be:
```
contracts/
    hts-precompile/
        safe/
            examples/
                SafeOperations.sol
            SafeHTS.sol
            HederaResponseCodes.sol
        IHederaTokenService.sol
        HederaTokenService.sol
        HederaResponseCodes.sol
    ...
``` 
or maybe placing the SafeHTS in a separate `libraries` directory i.e.
```
contracts/
    ...
libraries/
    examples/
        SafeOperations.sol
    SafeHTS.sol
    HederaResponseCodes.sol
```

Any thoughts?

Reply:
Let's go with the first option for now.
However, there should be code reuse. No need to repeat files like `HederaResponseCodes.sol` if there's no differnce.
Is it possible to share those files between the safe and non-hardened files?

Reply:
Actually why can't we do the following? Where Safe is just the hardened version of `HederaTokenService.sol` and `TokenManage.sol`.
Basically just sub folders under examples and they contain their details and share necessary files
The only thing outside of the examples should be the files that are actually on the network
```
contracts/
    hts-precompile/
        examples/
            safe/
                SafeOperations.sol
                SafeHTS.sol
            token-create/
                HederaTokenService.sol
            token-manage/
                TokenManage.sol
            HederaResponseCodes.sol
        IHederaTokenService.sol
    ...
```